### ISSUE TITLE
Importing onnx in macOS Catalina aborts if TensorFlow imported first

### ISSUE THREAD

**Issue by blake-qeexo** [onnx/onnx#2940]
# Bug Report

### Is the issue related to model conversion?
No
### Describe the bug
 If `tensorflow` is already imported, trying to `import onnx` throws this exception:
```
[libprotobuf ERROR google/protobuf/descriptor_database.cc:394] Invalid file descriptor data passed to EncodedDescriptorDatabase::Add().
[libprotobuf FATAL google/protobuf/descriptor.cc:1356] CHECK failed: GeneratedDatabase()->Add(encoded_file_descriptor, size): 
libc++abi.dylib: terminating with uncaught exception of type google::protobuf::FatalException: CHECK failed: GeneratedDatabase()->Add(encoded_file_descriptor, size): 
```

Importing `onnx` then `tensorflow` works correctly and does not abort.

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 16.04*): macOS 10.15.6
- ONNX version (*e.g. 1.7*):  1.7.0
- Python version: 3.8.5
- GCC/Compiler version (if compiling from source): N/A
- CMake version: 3.18.1
- Protobuf version: 3.12.2
- Visual Studio version (if applicable): N/A

### Reproduction instructions
Import `onnx` after `tensorflow`.

```
python3 -c 'import tensorflow, onnx'
```

### Expected behavior
For the imports to succeed.

**Comment by jcwchen** [comment]
Hi @blake-qeexo,

It seems like a common issue for Catalina.
I have the same version of protobuf (3.12.2) and onnx (1.7.0) as yours on my Mac, and it can work on my machine. (but I am using Mac 10.13.6)
According to https://github.com/tensorflow/tensorflow/issues/33746, maybe you can try to pip upgrade, pip reinstall all related packages, and see whether it can be imported successfully. 

Something like:
```
pip3 install --upgrade pip
pip3 uninstall tensorflow protobuf onnx && pip3 install tensorflow protobuf onnx
```

**Comment by TomWildenhain-Microsoft** [comment]
I'm also having this issue on MacOS Catalina.  I used a clean virtualenv and installed tensorflow and onnx (after upgrading pip and setuptools).  Still happened.  The crash occurs in __init__.py at the line `from .onnx_cpp2py_export import ONNX_ML` and is likely due to Catalina's new security requirements for dylibs.  https://www.shh.sh/2020/01/04/python-abort-trap-6.html

**Comment by pluradj** [comment]
I'm seeing the same issue on MacOS Catalina.

I'm trying to learn more about how the wheel file is generated. I don't understand how or why `onnx/.dylibs/libprotobuf.22.dylib` is packaged in the [`onnx-1.7.0-cp37-cp37m-macosx_10_9_x86_64.whl`](https://pypi.org/project/onnx/#files) if having protobuf and protobuf-compiler installed is an [ONNX prerequisite](https://github.com/onnx/onnx/blob/master/README.md#linux-and-macos).

Does anybody have insight on this?

**Comment by jcwchen** [comment]
Thanks for the useful link from @TomWildenhain-Microsoft. 
After updating to Catalina, I can reproduce this error.
A workaround would be using an older version of protobuf. Something like:
```
pip3 uninstall tensorflow protobuf onnx && pip3 install protobuf==3.11.3 tensorflow onnx
```
It should be able to solve most of issues on Catalina right now.
I will dig deeper into this to see what else we can do from onnx side.

Thanks.

**Comment by jcwchen** [comment]
Hi everyone,

I can import onnx successfully after import tensorflow by installing onnx from source or even from the release commit.
Moreover, it works normally if I install onnx by the wheel produced from the master onnx repo.
To fix this issue in pip wheel, we will review the process of building wheel and might release a small version of onnx in the future.

If you urgently need to use the latest protobuf on your Mac, you can try to rebuild onnx from source or the release commit (0c070abb0c40fec649f81a73a75b0098662ec486) of onnx 1.7.0. Sorry for the inconvenience.

**Comment by askhade** [comment]
This seems to be a known issue with tensorflow on catalina. 

Suggested workaround is to use an older version of protobuf (3.11.xx and lower ).
See related issues : https://github.com/tensorflow/tensorboard/issues/2985 and https://github.com/tensorflow/tensorflow/issues/33746

Building ONNX from source also works. 

Closing this issue now.

**Comment by dreamibor** [comment]
To be clear, the key here is to compile and install ONNX with a much lower version of protobuf. For example:
```conda install -c conda-forge protobuf numpy

git clone https://github.com/onnx/onnx.git
cd onnx
git submodule update --init --recursive
python setup.py install
```
Then we can install TensorFlow as normal, which will also upgrade the version of protobuf.

`conda install -c conda-forge protobuf` installs protobuf==3..4.0 on my Mac and it's more or less equivalent to 
`pip install protobuf==3..4.0`.

The information here is not clear, install from source with a higher version of protobuf won't work.

**Comment by jcwchen** [comment]
Hi @dreamibor, I believe this issue has been resolved by ONNX 1.8.1. Thank you.
